src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -49,22 +61,59 @@ public PersonCard(ReadOnlyPerson person, int displayedIndex) {
     }
 
     /**
+     * Get a color for a tag
+     */
+    private static String colorGetterForTag(String tagValue) {
+        int colorCode;
+        boolean usedUpAllColors = true;
+        for (int i = 0; i < colors.length; i++) {
+            if (usedColors[i] == 0) {
+                usedUpAllColors = false;
+                break;
+            }
+        }
+        if (usedUpAllColors) {
+            for (int j = 0; j < colors.length; j++) {
+                usedColors[j] = 0;
+            }
+        }
+
+        if (!tagColors.containsKey(tagValue)) {
+            do {
+                colorCode = random.nextInt(colors.length);
+            } while(usedColors[colorCode] == 1);
+            usedColors[colorCode] = 1;
+            tagColors.put(tagValue, colors[colorCode]);
+        }
+
+        return tagColors.get(tagValue);
+    }
+
+    /**
      * Binds the individual UI elements to observe their respective {@code Person} properties
      * so that they will be notified of any changes.
      */
     private void bindListeners(ReadOnlyPerson person) {
         name.textProperty().bind(Bindings.convert(person.nameProperty()));
         phone.textProperty().bind(Bindings.convert(person.phoneProperty()));
+        birthday.textProperty().bind(Bindings.convert(person.birthdayProperty()));
         address.textProperty().bind(Bindings.convert(person.addressProperty()));
         email.textProperty().bind(Bindings.convert(person.emailProperty()));
         person.tagProperty().addListener((observable, oldValue, newValue) -> {
             tags.getChildren().clear();
-            person.getTags().forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
+            initTags(person);
         });
     }
 
+    /**
+     * Distribute colors for tags
```

Comment:
Coding standard violation:exclamation: The first statement of the header comment does not follow the prescribed style. See point 4, 3rd bullet point of the [[Java Coding Standard: Comments](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#comments)]